test/Constraints/swift_to_c_pointer_conversions.swift.gyb

```diff
@@ -0,0 +1,116 @@
+// %target-swift-frontend(mock-sdk: %clang-importer-sdk-nosource -I %t) -import-objc-header %S/Inputs/overloaded_async_method.h %s -typecheck -verify
+
+// RUN: %empty-directory(%t)
+
+// RUN: %gyb %s -o %t/swift_to_c_pointer_conversions.swift
+
+// RUN: %line-directive %t/swift_to_c_pointer_conversions.swift -- %target-swift-frontend(mock-sdk: %clang-importer-sdk-nosource -I %t) -import-objc-header %S/Inputs/c_pointer_conversions.h %t/swift_to_c_pointer_conversions.swift -typecheck -verify
+
+func test_raw_ptr(ptr: UnsafeRawPointer) {
+  char_ptr_func(ptr) // expected-error {{}}
+  unsigned_char_ptr_func(ptr) // expected-error {{}}
+
+  const_char_ptr_func(ptr) // Ok
+  const_unsigned_char_ptr_func(ptr) // Ok
+
+  % for Size in ['16', '32', '64']:
+    int_${Size}_ptr_func(ptr) // expected-error {{}}
+    uint_${Size}_ptr_func(ptr) // expected-error {{}}
+
+    const_int_${Size}_ptr_func(ptr) // expected-error {{}}
+    const_uint_${Size}_ptr_func(ptr) // expected-error {{}}
+  % end
+}
+
+func test_mutable_raw_pointer(ptr: UnsafeMutableRawPointer) {
+  char_ptr_func(ptr) // Ok
+  unsigned_char_ptr_func(ptr) // Ok
+
+  const_char_ptr_func(ptr) // Ok
+  const_unsigned_char_ptr_func(ptr) // Ok
+
+  % for Size in ['16', '32', '64']:
+    int_${Size}_ptr_func(ptr) // expected-error {{}}
+    uint_${Size}_ptr_func(ptr) // expected-error {{}}
+
+    const_int_${Size}_ptr_func(ptr) // expected-error {{}}
+    const_uint_${Size}_ptr_func(ptr) // expected-error {{}}
+  % end
+}
+
+%for TestPtrSize in ['16', '32', '64']:
```

Comment:
I think we should also have tests for passing `Unsafe[Mutable]Pointer<[U]Int8>` as an argument:
to `char *` - ok
to `int16-64` - error

Reply:
Ugh, for some reason I have completely ignored 8-bit integers...

Reply:
Added `8` bit size to both tests blocks, so everything should be covered now.

Reply:
ok, looks good. I think we should test that `UP<[U]Int8>` cannot be converted to other integer sizes, e.g. `UP<[U]Int32>`. Just to be clear that it is intentionally not supported. Someone might expect that we meant to support that case, but it's only safe to do that conversion in one direction. We're only giving [U]Int8 special aliasing powers on the C side of the call.

Reply:
That's what the tests down the line do e.g. `uint_${TestPtrSize}_ptr_func(${pointer})`, gyb doesn't make it easy to read but `Int8` is tested against all the signed and unsigned integer types of 16, 32, 64 and all `*_char_ptr_func` test it against Int8 and UInt8 pointers at the top of the test function.

Reply:
Oh right... I was just looking at the diff. So I can't think of anything else to test then.

Reply:
No worries!